File: app/src/main/java/com/example/shreyastest/ViewModel/MainActivityVm.java
package com.example.shreyastest.ViewModel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;

import com.example.shreyastest.BeanClasses.Images;
import com.example.shreyastest.Repository.MainActivityRepo;

import java.util.List;

public class MainActivityVm extends AndroidViewModel {

    private MainActivityRepo mainActivityRepo;

    public MainActivityVm(@NonNull Application application) {
        super(application);
        mainActivityRepo = MainActivityRepo.getInstance(application);
    }

    public MutableLiveData<List<Images>> getImagesMutableLiveData(){
        return mainActivityRepo.mutableLiveData;
    }

    public void setDispose(){
        mainActivityRepo.onDisponse();
    }

    public void searchWord(String str){
        mainActivityRepo.getData(str);
    }

}

File: app/src/main/java/com/example/shreyastest/viewmodel/MainActivityVm.kt
package com.example.shreyastest.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import com.example.shreyastest.beanclasses.Images
import com.example.shreyastest.repository.MainActivityRepo

class MainActivityVm(application: Application) : AndroidViewModel(application) {
    private val mainActivityRepo: MainActivityRepo
    val imagesMutableLiveData: MutableLiveData<List<Images>>
        get() = mainActivityRepo.mutableLiveData

    fun setDispose() {
        mainActivityRepo.onDisponse()
    }

    fun searchWord(str: String?) {
        mainActivityRepo.getData(str)
    }

    init {
        mainActivityRepo = MainActivityRepo.getInstance(application)
    }
}